src/intro_movie_scenes_lib.c

```diff
@@ -0,0 +1,528 @@
+})
+
+void Task_IntroMovie_WindowPanEffect(SysTask *task, void *pVoid) {
+    IntroMovieBgWindowAnim *data = (IntroMovieBgWindowAnim *)pVoid;
+    ++data->counter;
+    int x1 = scalePos(data->param.x1Start, data->param.x1End, data->counter, data->duration);
+    int y1 = scalePos(data->param.y1Start, data->param.y1End, data->counter, data->duration);
+    int x2 = scalePos(data->param.x2Start, data->param.x2End, data->counter, data->duration);
+    int y2 = scalePos(data->param.y2Start, data->param.y2End, data->counter, data->duration);
+    data->x1 = x1;
+    data->y1 = y1;
+    data->x2 = x2;
+    data->y2 = y2;
+    if (data->counter >= data->duration) {
+        SysTask_Destroy(data->task);
+        data->task = NULL;
+        data->finished = TRUE;
+        data->active = FALSE;
+    }
+    IntroMovie_SetBgWindowsPosition(x1, y1, x2, y2, data->whichScreen);
+}
+
+#undef scalePos
+
+void IntroMovie_WindowsOn_SetInsideOutsidePlanes(int winIn, int winOut, u8 topScreenEffect, u8 bottomScreenEffect, int whichScreen) {
+    if (whichScreen) {
+        GX_SetVisibleWnd(3);  // both
+        G2_SetWnd0InsidePlane(winIn, topScreenEffect);
+        G2_SetWnd1InsidePlane(winIn, topScreenEffect);
+        G2_SetWndOutsidePlane(winOut, topScreenEffect);
+    } else {
+        GXS_SetVisibleWnd(3);  // both
+        G2S_SetWnd0InsidePlane(winIn, bottomScreenEffect);
+        G2S_SetWnd1InsidePlane(winIn, bottomScreenEffect);
+        G2S_SetWndOutsidePlane(winOut, bottomScreenEffect);
+    }
+}
+
+void IntroMovie_SetBgWindowsPosition(int x1, int y1, int x2, int y2, int whichScreen) {
+    if (x1 == 0 && x2 == 0xFF) {
+        if (whichScreen) {
+            G2_SetWnd1Position(0, y1, 1, y2);
+            G2_SetWnd0Position(1, y1, 0, y2);
+        } else {
+            G2S_SetWnd1Position(0, y1, 1, y2);
+            G2S_SetWnd0Position(1, y1, 0, y2);
+        }
+    } else {
+        if (whichScreen) {
+            G2_SetWnd0Position(x1, y1, x2, y2);
+        } else {
+            G2S_SetWnd0Position(x1, y1, x2, y2);
+        }
+    }
+}
+
+
+BgConfig *IntroMovie_GetBgConfig(IntroMovieOvyData *data) {
+    return data->bgConfig;
+}
+
+IntroMovieBgLinearAnims *IntroMovie_GetBgLinearAnimsController(IntroMovieOvyData *data) {
+    return &data->bgAnimCnt;
+}
+
+BOOL IntroMovie_GetIntroSkippedFlag(IntroMovieOvyData *data) {
+    return data->introSkipped;
+}
+
+int IntroMovie_GetTotalFrameCount(IntroMovieOvyData *data) {
+    return data->totalFrameCount;
+}
+
+void IntroMovie_InitBgAnimGxState(IntroMovieOvyData *data) {
+    G2_BlendNone();
+    G2S_BlendNone();
+    GX_SetVisibleWnd(0);
+    GXS_SetVisibleWnd(0);
+    for (int i = 0; i < 8; ++i) {
+        BgSetPosTextAndCommit(data->bgConfig, i, BG_POS_OP_SET_X, 0);
+        BgSetPosTextAndCommit(data->bgConfig, i, BG_POS_OP_SET_Y, 0);
+    }
+}
+
+void IntroMovie_BeginCircleWipeEffect(IntroMovieOvyData *data, int kind, BOOL isTopScreen, int duration) {
+    int winIn = 0x1F;
```

Comment:
whwt is a winIn?

Reply:
not sure, blame nintendo for the name

Reply:
we're not nintendo, pls can we rename a lot of these things

Reply:
fuck no

Reply:
too tired and busy to argue this so I'll probably just fix it myself